content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to your models, result objects, queries, or to Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To use a Prisma Client extension, you must create an associated _extended client_. An extended client is a forked variant of the standard Prisma Client with an extension applied. The standard client does not change.
+
+You can use [multiple independent extensions at the same time](#multiple-extensions). You can run an extension in its own isolated extended client, or combine it with other extensions in an extended client.
+
+You can also [share your Prisma Client extensions](/concepts/components/prisma-client/client-extensions/shared-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+### Client instances
+
+Extended clients interact with each other, and with the standard client, as follows:
+
+- Each extended client operates independently in an isolated instance.
+- Extended clients cannot conflict with each other, or with the standard client.
+- All extended clients and the standard client communicate with the same [Prisma query engine](/concepts/components/prisma-engines/query-engine).
+- All extended clients and the standard client share the same connection pool.
+
+#### Examples of isolated client instances
+
+Example uses of isolated client instances include the following:
+
+- Implement row-level security (RLS), where each http request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
```

Comment:
I think this description might be confusing for beginners. It talks about an HTTP request without further context (there might be Prisma applications that are not using HTTP, albeit probably the exception). I'm assuming this is in the context of a HTTP web server but this should be clarified IMO. I also don't fully understand the notion of "each http request has its own client".

Reply:
```suggestion
- Implement row-level security (RLS), where each HTTP request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
```